motoko/dao/src/Types.mo

```diff
@@ -0,0 +1,65 @@
+import Result "mo:base/Result";
+import Trie "mo:base/Trie";
+import Int "mo:base/Int";
+import List "mo:base/List";
+import Principal "mo:base/Principal";
+
+module {
+  public type Result<T, E> = Result.Result<T, E>;
+  public type Account = { owner : Principal; tokens : Tokens };
+  public type Proposal = {
+    id : Nat;
+    votes_no : Tokens;
+    voters : List.List<Principal>;
```

Comment:
Besides `Err` vs `err`, this is another difference in the did file. Rust is using `vec principal`, and Motoko uses `opt record { principal; list }`.

The Rust implementation can use the same ic-repl scripts, except for the above differences.